File: is_palindrome/is_palindrome.go
package main

import (
 "fmt"
)

func main() {
	//check function
	if(isPalindrome("mam")){
		fmt.Println("Palindrome checker worked!")
	}else{
		fmt.Println("Oops! there is a problem")
	}
 }

func isPalindrome(s string) bool {
	mid := len(s) / 2
	last := len(s) - 1
	for i := 0; i < mid; i++ {
		if s[i] != s[last-i] {
			return false
		}
	}
	return true
}

File: is_palindrome/is_palindrome.cs
public class is_palindrome
{
	public bool isPalindrome(string str)
	{
		for (int i = 0, j = str.Length - 1; i < str.Length / 2; i++, j--)
		{
			if (str[i] != str[j])
				return false;
		}
		return true;
	}
}
